internal/pkg/vm/internal/vmcontext/invocation_context.go

```diff
@@ -71,7 +73,9 @@ func (ctx *invocationContext) invoke() interface{} {
 	// 6. create target state handle
 
 	// assert from address is an ID address.
-	runtime.Assert(ctx.msg.from.Protocol() == address.ID)
+	if ctx.msg.from.Protocol() != address.ID {
+		runtime.Abortf(exitcode.ErrForbidden, "Sender address MUST be an ID address at invocation time")
```

Comment:
If the ctx is supposed to have the sender resolved already, it would be more appropriate to assert or return an explicit error here: it indicates a programming error that's probably not repeatable in other implementations.